contracts/eth/nearbridge/contracts/NearBridge.sol

```diff
@@ -88,13 +92,18 @@ contract NearBridge is INearBridge, AdminControlled {
         payable(msg.sender).transfer(amount);
     }
 
-    function challenge(address payable receiver, uint signatureIndex) public override pausable(PAUSED_CHALLENGE) {
+    function challenge(address payable receiver, uint signatureIndex)
+        external
+        override
+        nonReentrant
```

Comment:
No need `nonReentrant`. It is acceptable at this point, since this function is not called often.

Reply:
Exactly because this state variable will block reentrancy by default: 

`lastValidAt = 0;`